backend/routes/admin.js

```diff
@@ -1,93 +1,105 @@
-          }
-    },
-  ])
-  .then(function(found) {
-    res.json(found);
-  })
-  .catch(function(err) {
-    res.send(err);
-  });
+      {
+        $project: {
+          candidateName: "$Organization Name",
+          numReviews: { $size: "$applicationReviews" },
+          avgRating: { $round: [{ $avg: "$applicationReviews.rating" }, 2] }
+        }
+      }
+    ])
+    .then(function(found) {
+      res.json(found);
+    })
+    .catch(function(err) {
+      res.send(err);
+    });
 });
 
-
 router.get("/", function(req, res) {
-
-
   //Note: Firebase only returns 1000 users at a time so you have to do it in batches
   //https://firebase.google.com/docs/auth/admin/manage-users#list_all_users
   function listAllUsers(nextPageToken) {
-
-    let users = []
+    const users = [];
     // List batch of users, 1000 at a time.
-    firebaseAdmin.auth().listUsers(1000, nextPageToken)
-      .then(listUsersResult => {
-            listUsersResult.users.forEach(function(userRecord) {
-              users.push(userRecord)
-            //console.log('user', userRecord);
-          });
-          res.json(listUsersResult.users);
-          if (listUsersResult.pageToken) {
-            // List next batch of users.
-            listAllUsers(listUsersResult.pageToken);
-          }
+    firebaseAdmin
+      .auth()
+      .listUsers(1000, nextPageToken)
+      .then((listUsersResult) => {
+        listUsersResult.users.forEach(function(userRecord) {
+          users.push(userRecord);
+          //console.log('user', userRecord);
+        });
+        res.json(listUsersResult.users);
+        if (listUsersResult.pageToken) {
+          // List next batch of users.
+          listAllUsers(listUsersResult.pageToken);
+        }
       })
       .catch(function(error) {
-        console.log('Error listing users:', error);
+        console.log("Error listing users:", error);
       });
+  }
+  listAllUsers();
+});
 
-      console.log("Hello there", users);
+function randomPassword(length) {
+  const chars =
+    "abcdefghijklmnopqrstuvwxyz!@#$%^&*()-+<>ABCDEFGHIJKLMNOP1234567890";
+  let pass = "";
+  for (let x = 0; x < length; x++) {
+    const i = Math.floor(Math.random() * chars.length);
+    pass += chars.charAt(i);
   }
-  listAllUsers()
+  return pass;
+}
 
-});
+async function createUser(user) {
+  const password = randomPassword(Math.floor(Math.random() * 5 + 8));
+  try {
+    const userRecord = await firebaseAdmin.auth().createUser({
+      displayName: user.name,
+      email: user.email,
+      password
+    });
+    const link = await firebaseAdmin
+      .auth()
+      .generatePasswordResetLink(user.email);
+    await sendWelcomeEmail(user, link);
+    return userRecord;
+  } catch (e) {
+    return Promise.reject(e);
+  }
+}
 
+router.post("/create-user", async function(req, res) {
+  try {
+    const user = await createUser(req.body);
+    res.json(user);
+  } catch (e) {
+    res.status(409).send(e);
```

Comment:
Is the 409 because we're assuming the only source of error is an already-existing email? If there are other sources of error then we might want a more generic status code.